cpp/src/parquet/column_reader.cc

```diff
@@ -442,8 +442,10 @@ std::shared_ptr<Page> SerializedPageReader::NextPage() {
                           header.repetition_levels_byte_length, &levels_byte_len)) {
         throw ParquetException("Levels size too large (corrupt file?)");
       }
-      page_buffer = DecompressIfNeeded(std::move(page_buffer), compressed_len,
-                                       uncompressed_len, levels_byte_len);
+      if (is_compressed) {
+        page_buffer = DecompressIfNeeded(std::move(page_buffer), compressed_len,
```

Comment:
could you add a comment here.  The logic reads strangely to me because I would assume is_compressed is somehow accounted for in DecompressIfNeeded.